terraform/ci/main.tf

```diff
@@ -29,30 +33,65 @@ resource "google_project_service" "services" {
   disable_on_destroy = false
 }
 
-// IAM roles needed to run tests.
-resource "google_project_iam_member" "gh_access_acc_iam" {
-  for_each = toset(var.ci_iam_roles)
-
+resource "google_artifact_registry_repository" "artifact_repository" {
   project = var.project_id
 
-  role   = each.key
-  member = module.github_ci_infra.service_account_member
-}
+  location      = var.region
+  repository_id = var.registry_repository_id
+  description   = "Container registry for docker images."
+  format        = "DOCKER"
 
-module "github_ci_infra" {
```

Comment:
`github_ci_infra` creates wif and artifact registry, but now wif should be handled by [infra](https://github.com/abcxyz/infra/tree/main/terraform/workload-identity/github). So we remove this part, and create AR in CI, and use the `wif` created by infra is an input to bind with AR.

Reply:
I think we have not moved jvs to the new strcuture handled by [infra](https://github.com/abcxyz/infra/tree/main/terraform/workload-identity/github). As you can see, there is no wif related repo to JVS https://github.com/abcxyz/infra/tree/main/terraform/workload-identity/github/abcxyz

Reply:
+1, this is not safe to remove since https://github.com/abcxyz/infra/blob/main/terraform/deployment-resources/jvs-legacy/ci/main.tf#L6 does not pin the commit sha, so if anyone makes a change in infra, the wif will be removed, and we don't have one to replace.

Reply:
This PR is just show idea of the CI module looks like that's why it's in draft. I won't merge this until I have everything needed is set up in infra.